spec/cli_spec.rb

```diff
@@ -0,0 +1,175 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+describe Policial::CLI do
+  subject { described_class.new }
+
+  before(:each) { ENV['GITHUB_ACCESS_TOKEN'] = '1234' }
+  after(:each) { ENV['GITHUB_ACCESS_TOKEN'] = nil }
+
+  describe '#investigate' do
+    context 'when GITHUB_ACCESS_TOKEN environment variable is not set' do
+      it 'aborts with exit status code 1' do
+        ENV['GITHUB_ACCESS_TOKEN'] = nil
+
+        output = capture(:stderr) do
+          expect(-> { subject.investigate('volmer/policial', 1) })
+            .to exit_with_code(1)
+        end
+
+        expect(output).to eq(
+          "You must set the GITHUB_ACCESS_TOKEN environment variable.\n"
+        )
+      end
+    end
+
+    context 'when repo is not specified' do
+      context 'within a git repository folder' do
+        context 'that has no GitHub remote' do
+          it 'aborts with exit status code 1' do
+            allow(Open3).to receive(:popen3)
```

Comment:
If most examples will stub this then this could go on a top-level `before` block and we could just change the expected returned values in each context.